[ CHEC-753] Adding inline alert version

kvisca opened the pull request:
- Adding the inline alert version.
- Adding the new square icons.
- Adding option to disable close button
- Updating story to reflect new options and inline versions.  

![Screen Shot 2020-07-02 at 5 47 08 PM](https://user-images.githubusercontent.com/36721153/86412013-1a3efb80-bc8c-11ea-93e7-4b8eed405eb7.png)

ScopeyNZ commented on src/stories/components/BaseAlert.stories.mdx:
```diff
@@ -93,4 +103,77 @@ import BaseAlert from '../../components/BaseAlert.vue';
+          if (variant == 'info') {
+            return 'info-square';
+          }
+          return 'info-square';
```
What do you think about making this stuff part of the component - as a `show-icon` prop?

ScopeyNZ commented on src/components/BaseAlert.vue:
```diff
@@ -54,21 +62,25 @@ export default {
     @apply ml-4 p-2 bg-transparent rounded text-white cursor-pointer outline-none;
-
+    margin-top: -2rem;
+    margin-bottom: -2rem;
```
There's `@apply -my-8` for this too.

kvisca replied:
This is so when the close button is removed the container doesn't become smaller than it should be.

ScopeyNZ commented on src/components/BaseAlert.vue:
```diff
@@ -89,6 +101,55 @@ export default {
+      }
+
+    }
+    &.alert--warning{
```
See https://github.com/chec/ui-library/blob/ff95f6044827e3e1b163b1efa30b6774aa1553d2/src/components/BaseButton.vue#L239

robbieaverill commented:
Can you remove the `.DS_Store` file and add that filename to your global gitignore rules?

ScopeyNZ commented on src/components/BaseAlert.vue:
```diff
@@ -35,10 +38,37 @@ export default {
   computed: {
     classObject() {
-      return `alert--${this.variant}`;
+      return `alert--${this.variant} ${this.inline ? 'alert--inline' : ''} ${this.icon ? 'alert--icon' : ''}`;
```
This is fine - but for future reference you can write this as:

```suggestion
      return [`alert--${this.variant}`, {
        'alert--inline': this.inline,
        'alert--icon': this.icon,
      }];
```

I think it's a tiny bit more readable.

kvisca replied:
Updated

ScopeyNZ commented:
Nice. I was just about to merge but I remembered that we actually have the non-coloured variant in use on the new docs:

https://www.figma.com/file/ypYnoIU8yJfnhek3h2ZRvQ/Docs-Website?node-id=3%3A686

We'll need that as an option

Additionally, can you add a knob for the "inline" prop to the "Default" story?

kvisca commented:
> Additionally, can you add a knob for the "inline" prop to the "Default" story?

Fixed the issue where info displayed in incorrect color and updated the story.

ScopeyNZ commented:
🎉
